core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -935,9 +937,42 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     self.sendInterruptSignalToAllChildren(_children)
 
     val k = self._asyncContWith
+    self._asyncContWith = AsyncContWith.`null`
+
+    val callback = k.callback
+
+    // Not async interruption or uninterruptible async
+    if (callback eq null) return
+
+    k.onInterrupt match {
+      // No interrupt handler
+      case null => callback(Exit.Failure(cause))
+
+      // Shortcut cases where the interruption is a simple suspended function (most common)
+      case sync: Sync[Any] =>
```

Comment:
Kind of dirty but in most of our usages of `ZIO.asyncInterrupt` the interruption handler is `ZIO.succeed`